src/components/VideoDetail/VideoDetail.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import styled from 'styled-components';
+import './VideoDetail.styles.css';
+import { useVideoContext } from '../../state/VideosContext';
+import { useThemeContext } from '../../state/ThemeContext';
+
+const VideoDetailsContainer = styled.div`
+    display: grid;
+    grid-template-columns: 3fr 1fr;
+    grid-gap: 20px;
+    position: relative;
+`;
+
+const VideoPlayer = styled.iframe`
+    min-width: 100%;
+    height: 600px;
+`;
+
+function VideoDetail(){
+    const { videoList, state, dispatch } = useVideoContext();
+    const { id, title, description } = state;
+    const { theme } = useThemeContext();
+
+    const videoSrc = `https://www.youtube.com/embed/${id}`;
+    const getVideoList = (item) => {
+        return (item.items.map((videoItem, index) => {
+            const { id, snippet } = videoItem;
+            const { title, thumbnails, description } = snippet;
+            
+              return {
+                id: id.videoId,
+                title: title,
+                thumbnail: thumbnails.medium.url,
+                description: description,
+                shortthumbnail: thumbnails.default.url,
+              };
+        }));
+      }
+
+    return(
+        <VideoDetailsContainer style={{color: theme.fontcolor, backgroundColor: theme.backgroundcolor }}>
```

Comment:
Instead of adding inline properties, why not to pass the theme as an argument and then do the validation of colors using the props in the styled component: https://styled-components.com/docs/basics#adapting-based-on-props